assets/context.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+from collections import namedtuple
+
+import discord
+from discord.ext import commands
+
+from cogs.utils import misc, check
+from cogs.utils.embeds import ConfirmEmbed, ConfirmEmbedAnother
+
+argument = namedtuple("Usage", ["required", "text"])
+example = namedtuple("Examples", ["args", "text"])
+sub_desc = namedtuple("subcommands", "desc")
+others = namedtuple("others", ["name", "text"])
+usage_info = namedtuple("usage_information", ["command", "value"])
+keyword = namedtuple("keywords", ["keyword", "desc"])
+
+
+class InvokeError(commands.CommandError):
+    def __init__(self, message):
+        self.message = message
+
+
+class Context(commands.Context):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.InvokeError = InvokeError
+
+    async def confirm(self, title, content, color=misc.random_color()):
+        instance = ConfirmEmbed(self, content, color=color, title=title)
+        confirmation = await instance.start()
+        return confirmation
+
+    async def confirm_another(self, user, title, content, color=misc.random_color()):
+        instance = ConfirmEmbedAnother(ctx=self, user=user, content=content,
+                                       color=color, title=title, dm=True)
+        confirmation = await instance.start()
+        return confirmation
+
+    async def get_response(self, question, check_func=None, delete=True):
+        def validate_response(message):
+            if message.author == self.author and message.channel == self.channel and \
+               True if check_func is None else check_func(message.content):
+                return True
+            else:
+                return False
+
+        question_msg = await self.send(content=question)
+        try:
+            msg = await self.bot.wait_for("message", check=validate_response, timeout=30)
+        except asyncio.TimeoutError:
+            return None
+
+        if delete is True:
```

Comment:
R1705: Unnecessary "else" after "return" + 
```suggestion
        if delete:
```